src/cpu/core_dynrec/operators.h

```diff
@@ -1967,6 +1967,152 @@ static uint32_t DRC_CALL_CONV dynrec_stosd_dword(uint32_t count,int32_t add_inde
 	return count_left;
 }
 
+static uint16_t DRC_CALL_CONV dynrec_scasb_word(uint16_t count, int16_t add_index,
+                                                PhysPt di_base) DRC_FC;
+static uint16_t DRC_CALL_CONV dynrec_scasb_word(uint16_t count,
+                                                int16_t add_index, PhysPt di_base)
+{
+	for (; count > 0; count--) {
+		const auto addr = di_base + reg_di;
+		const auto val  = mem_readb(addr);
```

Comment:
`addr` might be an `int` from the addition; the `mem_read` calls take in a `PhysPt` (`uint32_t`).  Maybe need a cast the addition?